hummingbot/market/market_base.pyx

```diff
@@ -170,7 +171,17 @@ cdef class MarketBase(NetworkIterator):
         :returns: Balance available for trading for a specific asset
         (balances used to place open orders are not available for trading)
         """
-        return self._account_available_balances.get(currency, s_decimal_0)
+        cdef:
+            object balance
+            dict restriction_setting
+        balance = self._account_available_balances.get(currency, s_decimal_0)
+        if global_config_map.get("asset_restriction_enabled").value:
+            restriction_setting = {curr: bal for curr, bal in global_config_map.get("asset_restriction").value}
+            if currency in restriction_setting:
+                balance = min(balance, restriction_setting.get(currency))
+            else:
+                balance = 0
```

Comment:
I think we shouldn't apply asset restriction in case of when there is no restriction specified for the asset.